models/domains_domains.php

```diff
@@ -220,30 +276,36 @@ public function updateNameservers($service_id, array $nameservers)
      */
     public function getExpirationDate($service_id, $format = 'Y-m-d H:i:s')
     {
-        Loader::loadModels($this, ['Services', 'ModuleManager']);
+        Loader::loadModels($this, ['Services', 'ModuleManager', 'Companies', 'Services']);
+
+        if (is_null($format)) {
+            $format = 'Y-m-d H:i:s';
+        }
 
         if (is_numeric($service_id)) {
             $service = $this->Services->get($service_id);
 
-            // Check if the domain already has an expiration date
-            $domain = $this->Record->select()
-                ->from('domains_domains')
-                ->where('service_id', '=', $service->id)
-                ->fetch();
-            $expiration_date = $domain->expiration_date ?? null;
-
             // Get the expiration date from the registrar
-            if (empty($expiration_date)) {
-                $expiration_date = $this->ModuleManager->moduleRpc(
-                    $service->package->module_id,
-                    'getExpirationDate',
-                    [$service_id, $format],
-                    $service->module_row_id ?? null
-                );
-            }
+            $expiration_date = $this->ModuleManager->moduleRpc(
```

Comment:
Previously, we check for the expiration in the following order:
1. Domains table
2. Registrar API
3. Services table

This PR changes it to
1. Registrar API
2. Domains table
3. Services table

Why?

Reply:
Now the updateRenewalDate() event is triggered when the serviced is renewed (not only when added) to update the renewal date stored in the domains table, that's why getExpirationDate() now prioritizes the Registrar API, otherwise getExpirationDate() would return the exact same date we already have in the domains table and thereof the date would not be updated.